php-checks/src/main/java/org/sonar/php/checks/DuplicateBranchImplementationCheck.java

```diff
@@ -40,20 +51,82 @@
 @SqaleConstantRemediation("10min")
 public class DuplicateBranchImplementationCheck extends AbstractDuplicateBranchCheck {
 
-  @Override
-  protected AstNodeType ifBranchNodeType() {
-    return PHPGrammar.STATEMENT;
+  public static final String KEY = "S1871";
+  private static final String MESSAGE = "This %s's code block is the same as the block for the %s on line %s.";
+
+  private static class Tuple {
+    Tree clause;
+    List<Tree> body;
+
+    public Tuple(Tree clause, List<StatementTree> body) {
+      this.clause = clause;
+      this.body = ImmutableList.<Tree>copyOf(body);
```

Comment:
@vilchik-elena Why do we need to copy the list? If it's because we need instance of Tree instead of StatementTree, we may change some method signatures to accept List<? extends Tree> instead of List<Tree>.